Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/MainActivity.kt

```diff
@@ -171,11 +175,58 @@ class MainActivity : AppCompatActivity(), HasAndroidInjector {
             }
         }
 
+        viewModel.coronaTestResult.observe(this) {
+            handleCoronaTestResult(it)
+        }
+
         if (savedInstanceState == null) {
             processExtraParameters()
         }
     }
 
+    private fun handleCoronaTestResult(coronaTestResult: CoronaTestResult) = when (coronaTestResult) {
+        is CoronaTestResult.InRecycleBin -> {
+            showRestoreCoronaTestConfirmation(coronaTestResult.recycledCoronaTest)
+            null
+        }
+        is CoronaTestResult.RestoreDuplicateTest ->
+            NavGraphDirections.actionToSubmissionDeletionWarningFragment(
+                coronaTestResult.restoreRecycledTestRequest
+            )
+        is CoronaTestResult.TestInvalid ->
+            NavGraphDirections.actionGlobalSubmissionTestResultInvalidFragment(
+                testType = coronaTestResult.test.type,
+                testIdentifier = coronaTestResult.test.identifier
+            )
+        is CoronaTestResult.TestNegative ->
+            NavGraphDirections.actionGlobalSubmissionTestResultNegativeFragment(
+                testType = coronaTestResult.test.type,
+                testIdentifier = coronaTestResult.test.identifier
+            )
+        is CoronaTestResult.TestPending ->
+            NavGraphDirections.actionSubmissionTestResultPendingFragment(
+                testType = coronaTestResult.test.type,
+                testIdentifier = coronaTestResult.test.identifier,
+                forceTestResultUpdate = true
+            )
+        is CoronaTestResult.TestPositive ->
+            NavGraphDirections.actionGlobalSubmissionTestResultKeysSharedFragment(
+                testType = coronaTestResult.test.type,
+                testIdentifier = coronaTestResult.test.identifier
+            )
+        is CoronaTestResult.TestRegistrationSelection ->
```

Comment:
I think this the only case that has to handled here. 
- in case the test needs to be restored i.e `RestoreDuplicateTest` -> then test selection screen has to be shown -> the rest of the flow depending on whether it is personal or family test  
- In case the test is new, or existing i.e  `TestRegistrationSelection` ->  then test selection screen has to be shown

any other case, the card is displayed in HomeScreen anyway and `no need` to open the respective result screen